conbench/entities/benchmark_result.py

```diff
@@ -957,7 +998,9 @@ class BenchmarkResultStatsSchema(marshmallow.Schema):
         # strings are/were allowed to be injected. TODO: be more strict.
         # Require users to provide a unit (non-zero length string)./
         required=True,
-        metadata={"description": "The unit of the data object (e.g. seconds, B/s)"},
+        metadata={
+            "description": "The unit of the measurement result (object (e.g. seconds, B/s)"
```

Comment:
docs update here will come in a separate patch

Reply:
We should copy docs improvements to [benchadapt's result object](https://github.com/conbench/conbench/blob/6ab7ff096c2dc31a49fb9b0078935858c11a7c59/benchadapt/python/benchadapt/result.py#L34-L37) as well